api/routes/login.js

```diff
@@ -1,69 +1,105 @@
 const express = require('express');
+const bodyparser = require('body-parser');
+const cookieParser = require('cookie-parser');
+const User = require('../models/user');
+const { auth } = require('../middleware');
 const path = require('path');
-const jwt = require('jsonwebtoken');
-const withAuth = require('../middleware');
-const User = require('../models/User');
 
-const {JWT_SECRET} = require('../config');
 
 const app = express();
-
-app.get('/', function (req, res) {
-  res.sendFile(path.join(__dirname, 'public', 'index.html'));
+// app use
+app.use(bodyparser.urlencoded({ extended: false }));
+app.use(bodyparser.json());
+app.use(cookieParser());
+
+app.use('/', express.static(path.join(__dirname, 'static')));
+
+// adding new user (sign-up route)
+app.post('/register', function(req, res) {
+    let newUser = new User(req.body);
+    User.findOne({ email: newUser.email }, function(err, user) {
+        if (user) return res.status(400).json({
+            auth: false,
+            message: "email exits"
+        });
+
+        newUser.save((err, doc) => {
+            if (err) {
+                return res.status(400).json({
+                    error: true,
+                    message: err.message
+                });
+            }
+            res.status(200).json({
+                success: true,
+                user: { // Control the output we don't want to return the password that would be bad
+                    _id: doc._id,
+                    email: doc.email,
+                    name: doc.name
+                }
+            });
+        });
+    });
 });
 
-app.get('/home', function (req, res) {
-  res.send('Welcome!');
-});
 
-app.get('/secret', withAuth, function (req, res) {
-  res.send('You are visiting a protected page.');
+// login user
+app.post('/login', function(req, res) {
+    let token = req.cookies.auth;
+    User.findByToken(token, (err, user) => {
```

Comment:
Would say this is a bit unconventional normally you just verify the token and that is enough.
It expires after a set amount of time and you are the only party that can sign and verify it with your secret.

No need to save it on the model.

If you do want to save it for security reason, giving you control to destroy sessions I would suggest a different collection.

But no need to change now this works for now.